plugin/plugin.go

```diff
@@ -21,6 +21,23 @@ const (
 	TaskMonitorPlugin
 )
 
+// ErrUnsupported is raised when a plugin is not supported on the system
+type ErrUnsupported struct {
```

Comment:
We should not need a type for the cause of unsupported errors. This error pattern is something we've found to be pretty problematic. If you want a nice error message, just wrap an unsupported error with a nice message.